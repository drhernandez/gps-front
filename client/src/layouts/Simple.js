import React from "react";
import { Container, Row, Col } from "shards-react";

const DefaultLayout = ({ children }) => (
  <Container fluid>
    <Row>
      <Col className="main-content p-0" tag="main" >
        {children}
      </Col>
    </Row>
  </Container>
);

export default DefaultLayout;
