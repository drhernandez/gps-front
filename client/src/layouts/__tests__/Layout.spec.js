import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux';
import { Router } from "react-router-dom";
import { createMemoryHistory } from 'history'
import configureStore from 'redux-mock-store';
const mockStore = configureStore([]);

import Default from "../Default"
import Simple from "../Simple"

let store;

describe("Layouts", () => {

  beforeEach(async () => {
    store = mockStore({
      userInfo: {
        id: 1
      },
    });
  });

  it("Simple snapshot", () => {
    const component = renderer.create(
      <Simple />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })

  it("Default snapshot", () => {
    const history = createMemoryHistory()
    const component = renderer.create(
      <Router history={history}>
        <Provider store={store}>
          <Default />
        </Provider>
      </Router>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})