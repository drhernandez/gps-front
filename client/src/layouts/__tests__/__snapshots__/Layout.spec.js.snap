// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layouts Default snapshot 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  >
    <aside
      className="main-sidebar px-0 col-12 open col-md-3 col-lg-2"
    >
      <div
        className="main-navbar"
      >
        <nav
          className="align-items-stretch bg-white flex-md-nowrap border-bottom p-0 navbar navbar-light"
        >
          <a
            className="w-100 mr-0 navbar-brand"
            href="#"
            style={
              Object {
                "lineHeight": "25px",
              }
            }
          >
            <div
              className="d-table m-auto"
            >
              <img
                alt="Logo"
                className="d-inline-block align-top mr-1"
                id="main-logo"
                src={Object {}}
                style={
                  Object {
                    "maxWidth": "25px",
                  }
                }
              />
              <span
                className="d-none d-md-inline ml-1"
              >
                S.G.S.
              </span>
            </div>
          </a>
          <a
            className="toggle-sidebar d-sm-inline d-md-none d-lg-none"
            onClick={[Function]}
          >
            <i
              className="material-icons"
              id="hoa"
            >
              
            </i>
          </a>
        </nav>
      </div>
      <div
        className="nav-wrapper"
      >
        <ul
          className="nav--no-borders flex-column nav"
        >
          <li
            className="nav-item"
          >
            <a
              aria-current={null}
              className="nav-link"
              href="/new-client"
              onClick={[Function]}
            >
              <div
                className="d-inline-block item-icon-wrapper"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<i class=\\"material-icons\\">person_add</i>",
                  }
                }
              />
              <span>
                Alta de cliente
              </span>
              
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-current={null}
              className="nav-link"
              href="/clients"
              onClick={[Function]}
            >
              <div
                className="d-inline-block item-icon-wrapper"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<i class=\\"material-icons\\">build</i>",
                  }
                }
              />
              <span>
                Administrar clientes
              </span>
              
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-current={null}
              className="nav-link"
              href="/tracking"
              onClick={[Function]}
            >
              <div
                className="d-inline-block item-icon-wrapper"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<i class=\\"material-icons\\">my_location</i>",
                  }
                }
              />
              <span>
                Tracking de clientes
              </span>
              
            </a>
          </li>
        </ul>
      </div>
    </aside>
    <main
      className="main-content p-0 col-sm-12 col-md-9 offset-md-3 col-lg-10 offset-lg-2"
    >
      <div
        className="main-navbar bg-white sticky-top"
      >
        <div
          className="p-0 container"
          style={
            Object {
              "maxWidth": "100%",
            }
          }
        >
          <nav
            className="d-flex flex-row-reverse  p-0 navbar navbar-light"
          >
            <ul
              className="border-left flex-row navbar-nav"
            >
              <div
                className="nav-item dropdown"
              >
                <a
                  aria-haspopup={true}
                  className="d-flex align-items-center h-100 text-nowrap px-3 dropdown-toggle nav-link"
                  onClick={[Function]}
                >
                  <span
                    className="d-none d-inline-block"
                  >
                    undefined undefined
                  </span>
                </a>
                <div
                  className="collapse dropdown-menu dropdown-menu-small dropdown-menu-right"
                  role="menu"
                  style={
                    Object {
                      "display": false,
                      "height": null,
                    }
                  }
                  tabIndex="-1"
                >
                  <a
                    className="dropdown-item"
                    href="/user-profile"
                    onClick={[Function]}
                    tabIndex="0"
                  >
                    <i
                      className="material-icons"
                    >
                      
                    </i>
                     Profile
                  </a>
                  <a
                    className="dropdown-item"
                    href="/edit-user-profile"
                    onClick={[Function]}
                    tabIndex="0"
                  >
                    <i
                      className="material-icons"
                    >
                      
                    </i>
                     Edit Profile
                  </a>
                  <div
                    className="dropdown-divider"
                    onClick={[Function]}
                    tabIndex="-1"
                  />
                  <a
                    className="text-danger dropdown-item"
                    href="/logout"
                    onClick={[Function]}
                    tabIndex="0"
                  >
                    <i
                      className="material-icons text-danger"
                    >
                      
                    </i>
                     Logout
                  </a>
                </div>
              </div>
            </ul>
            <nav
              className="nav"
            >
              <a
                className="nav-link nav-link-icon toggle-sidebar d-sm-inline d-md-inline d-lg-none text-center"
                href="#"
                onClick={[Function]}
              >
                <i
                  className="material-icons"
                >
                  
                </i>
              </a>
            </nav>
          </nav>
        </div>
      </div>
      <footer
        className="main-footer d-flex p-2 px-3 bg-white border-top"
      >
        <div
          className="container"
        >
          <div
            className="d-flex align-items-center h-100 row"
          >
            <span
              className="copyright ml-auto my-auto mr-2"
            >
              Copyright © 2019 DesignRevision
            </span>
          </div>
        </div>
      </footer>
    </main>
  </div>
</div>
`;

exports[`Layouts Simple snapshot 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  >
    <main
      className="main-content p-0 col"
    />
  </div>
</div>
`;
